### PR TITLE
DOC: Fix some references in document

### PR DESCRIPTION
**Pull request by ghost**
[skip cirrus] [skip azp] [skip actions]

### CODE REVIEW

File: doc/source/reference/c-api/types-and-structures.rst [hunk 1]
```diff
@@ -97,8 +97,7 @@ PyArray_Type and PyArrayObject
           /* version dependent private members */
       } PyArrayObject;
 
-   .. c:macro:: PyObject_HEAD
-
+   :c:macro:`PyObject_HEAD`
```
Comment (mattip) [inline doc/source/reference/c-api/types-and-structures.rst:100]: I don't think this is right, the correct place to point a reference to PyObject_HEAD is to the [PyObject_HEAD](https://docs.python.org/3/c-api/structures.html#c.PyObject_HEAD) in the cpython documentation. Is there a warning about a missing reference somewhere?

File: doc/source/user/basics.types.rst [hunk 2]
```diff
@@ -291,7 +291,7 @@ or when it checks specifically whether a value is a Python scalar. Generally,
 problems are easily fixed by explicitly converting array scalars
 to Python scalars, using the corresponding Python type function
 (e.g., :class:`int`, :class:`float`, :class:`complex`, :class:`str`,
-:class:`unicode`).
+``unicode`` (v2.x)).
```
Comment (mattip) [inline doc/source/user/basics.types.rst:294]: Python2 is long gone, I think we can remove this.

Comment (F3eQnxN3RriK) [reply doc/source/user/basics.types.rst]: Thanks, removed it.

File: doc/source/reference/c-api/types-and-structures.rst (hunk 1 above)
Comment (F3eQnxN3RriK) [reply doc/source/reference/c-api/types-and-structures.rst:100]: This reference points to cpython documentation(https://output.circle-artifacts.com/output/job/fc72e3e8-643a-471e-b9d6-c2bf1eda06f4/artifacts/0/doc/build/html/reference/c-api/types-and-structures.html#c.NPY_AO), with no warning about it.

**Comment by mattip** [comment]
Thanks @F3eQnxN3RriK. Note that each commit message must have the CI skip commands, otherwise it triggers CI.

**Pull request merged**